### ISSUE TITLE
[HOLD for payment 2025-01-02] [$250] Inconsistent when clicking 'get $250' when submitting expense

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#54022]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.75-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @dukenv0307 
**Slack conversation** (hyperlinked to channel name): [expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1733996083944379)

## Action Performed:
1. Login with the new user
2. Try to submit an expense
3. Choose type manual, enter a value and click on next button
4. Click to 'get $250' while the selection list content is empty
5. Verify that: it navigates to an external link
6. Search for some one
7. Click to 'get $250' at the bottom
8. Verify that: it navigates to Referral modal
## Expected Result:
When clicking on get $250 in both positions, they both navigate to Referral modal
## Actual Result:
Inconsistent when clicking 'get $250' when submitting expense
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

https://github.com/user-attachments/assets/425fb82a-4b43-4a61-85d1-16106dae523b

https://github.com/user-attachments/assets/80382704-fe01-403a-9845-72f96d008f7a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021868776434220055313</li>
        <li>Upwork Job ID: 1868776434220055313</li>
        <li>Last Price Increase: 2024-12-16</li>
<li>Automatic offers: </li>
<ul>
<li>dominictb | Contributor | 105414009</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @muttmuure</details>

**Comment by dvck27** [comment]
### Problem Summary
The "Get $250" button behaves inconsistently when submitting an expense. When the selection list is empty, clicking the button redirects to an external link instead of the Referral Modal. This inconsistency disrupts the user experience and can lead to confusion.

### Root Cause
In the file EmptySelectionListContent.tsx, the button is set to navigate to an external link when the selection list content is empty. This differs from the expected behavior, which should consistently direct users to the REFERRAL_DETAILS_MODAL.

### Proposed Solution
To ensure consistent navigation, I propose modifying the onPress action in EmptySelectionListContent.tsx to always navigate to the REFERRAL_DETAILS_MODAL route, regardless of whether the selection list is empty or populated. The updated code would look like this:


<TextLink
    onPress={() => {
        Navigation.navigate(
            ROUTES.REFERRAL_DETAILS_MODAL.getRoute(
                CONST.REFERRAL_PROGRAM.CONTENT_TYPES.SUBMIT_EXPENSE, 
                Navigation.getActiveRouteWithoutParams()
            )
        );
    }}
>
    {translate(`emptyList.${contentType}.subtitleText2`)}
</TextLink>


>>>This change ensures that clicking the "Get $250" button always opens the Referral Modal, providing a seamless and predictable user experience.

### Testing Scenarios
I plan to cover the following scenarios to verify the solution:

1. Empty Selection List:

When the selection list is empty and the "Get $250" button is clicked, it should open the Referral Modal.

2. Populated Selection List:

When the selection list has items and the "Get $250" button is clicked, it should also open the Referral Modal.

I will test these changes across all supported platforms (Android, iOS, web, and desktop) to ensure compatibility and prevent any regressions.

### Additional Considerations

I will verify that the navigation works correctly in both staging and production environments.

I'll ensure the changes align with Expensify's coding standards and contributing guidelines.

I am ready to collaborate and make any adjustments necessary to meet your requirements. Please let me know if there are any additional considerations or feedback on my proposed solution.

**Comment by dvck27** [comment]
Contributor details
Your Expensify account email: davidfdo@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01ea654577b365c6b9

**Comment by arosiclair** [comment]
Yes, opening the modal is the correct way. So @dominictb's proposal LGTM.

**Comment by muttmuure** [comment]
@dominictb $250 (Contributor)
@dukenv0307 $250 (C+)

**Comment by JmillsExpensify** [comment]
$250 approved for @dukenv0307

**Comment by muttmuure** [comment]
Added test rail case
